src/plugins/intel_gpu/include/intel_gpu/primitives/gemm.hpp

```diff
@@ -38,42 +38,114 @@ struct gemm : public primitive_base<gemm> {
     gemm(const primitive_id& id,
          const std::vector<input_info>& inputs,
          const data_types data_type,
-         const bool transpose_input0 = false,
-         const bool transpose_input1 = false,
+         const bool transpose_input0,
+         const bool transpose_input1,
          const float alpha = 1.0f,
          const float beta = 0.0f,
          const size_t input_rank = 4,
          const size_t weight_rank = 4,
          const padding& output_padding = padding())
         : primitive_base(id, inputs, {output_padding}, {optional_data_type{ data_type }}),
-          transpose_input0(transpose_input0),
-          transpose_input1(transpose_input1),
+          transpose_input0(transpose_input0 ? 1 : 0),
+          transpose_input1(transpose_input1 ? 1 : 0),
           alpha(alpha),
           beta(beta),
           input_rank(input_rank),
           weight_rank(weight_rank) {
         if (inputs.size() != 2 && inputs.size() != 3) {
             throw std::invalid_argument("Invalid inputs count - gemm expects either two or three inputs");
         }
+
+        auto get_tranaposed_order = [] (size_t rank, bool transposed) {
+            std::vector<int64_t> order(rank);
+            std::iota(order.begin(), order.end(), 0);
+            if (transposed)
+                std::swap(order[rank - 1], order[rank - 2]);
```

Comment:
Is this priimtive only supports transpose of last two dims' switch? If so could you describe it somewhere?

Reply:
This constructor is for normal `gemm` that supports transposition of the last two dims. A new constructor for arbitrary transposition is added below this.

Reply:
Can't we merge them?

Reply:
Okay, I understood. Could I do this refactoring task in a new PR?

Reply:
Doing refactoring in a separate PR is okay. My curiosity is whether it "can" be or not.